support-workers/src/typescript/lambdas/createSalesforceContactLambda.ts

```diff
@@ -0,0 +1,154 @@
+import type { CreateZuoraSubscriptionState } from '../model/createZuoraSubscriptionState';
+import { stageFromEnvironment } from '../model/stage';
+import type {
+	CreateSalesforceContactState,
+	WrappedState,
+} from '../model/stateSchemas';
+import {
+	createSalesforceContactStateSchema,
+	wrapperSchemaForState,
+} from '../model/stateSchemas';
+import { ServiceProvider } from '../services/config';
+import type { SalesforceContactRecords } from '../services/salesforce';
+import { SalesforceService } from '../services/salesforce';
+import { getSalesforceConfig } from '../services/salesforceClient';
+import { user } from '../test/fixtures/salesforceFixtures';
+import { asRetryError } from '../util/errorHandler';
+import { getIfDefined } from '../util/nullAndUndefined';
+
+const stage = stageFromEnvironment();
+const salesforceServiceProvider = new ServiceProvider(stage, async (stage) => {
+	const config = await getSalesforceConfig(stage);
+	return new SalesforceService(config);
+});
+
+export const handler = async (
+	state: WrappedState<CreateSalesforceContactState>,
+) => {
+	try {
+		console.info(`Input is ${JSON.stringify(state)}`);
+		const createSalesforceContactState = wrapperSchemaForState(
+			createSalesforceContactStateSchema,
+		).parse(state).state;
+		const serviceForUser = await salesforceServiceProvider.getServiceForUser(
+			state.state.user.isTestUser,
+		);
+		const contactRecords = await serviceForUser.createContactRecords(
+			createSalesforceContactState.user,
+			createSalesforceContactState.giftRecipient,
+		);
+		return {
+			...state,
+			state: createNextState(createSalesforceContactState, contactRecords),
+		};
+	} catch (error) {
+		throw asRetryError(error);
+	}
+};
+
+const createNextState = (
+	state: CreateSalesforceContactState,
+	contactRecords: SalesforceContactRecords,
+): CreateZuoraSubscriptionState => {
+	switch (state.product.productType) {
+		case 'Contribution':
+			return {
+				productSpecificState: {
+					productType: 'Contribution',
```

Comment:
Would it be worth using `state.product.productType` to avoid duplicating the value? Applies to each branch of this switch statement, and is consistent with how we're retrieving other fields from `state`.

Reply:
Ohh sorry I didn't spot that, yeah sounds like a future thing 👍